packages/yarnpkg-core/sources/Project.ts

```diff
@@ -112,6 +113,28 @@ export type InstallOptions = {
   skipBuild?: boolean,
 };
 
+const INSTALL_STATE_FIELDS = {
+  restoreInstallersCustomData: [
+    `installersCustomData`,
+  ] as const,
+
+  restoreResolutions: [
+    `accessibleLocators`,
+    `optionalBuilds`,
+    `storedDescriptors`,
+    `storedResolutions`,
+    `storedPackages`,
+    `lockFileChecksum`,
+  ] as const,
+};
+
+type RestoreInstallStateOpts = {
+  [key in keyof typeof INSTALL_STATE_FIELDS]?: boolean;
+};
+
+// Just a type that's the union of all the fields declared in `INSTALL_STATE_FIELDS`
```

Comment:
Would be a good idea to use `/** ... */` (jsdoc) type comments since they show up on hover in various IDEs.

Reply:
That's more to explain the convoluted type declaration; the hover definition doesn't have this problem:
![image](https://user-images.githubusercontent.com/1037931/98560873-853ab780-22a8-11eb-950f-075aad9d9cb1.png)